packages/client/src/__tests__/utils/tmpfile.test.ts

```diff
@@ -1,6 +1,7 @@
 import fs from 'fs/promises';
 import { withTempFile } from '../../utils/tmpfile';
 
+/* eslint-disable @typescript-eslint/require-await */
```

Comment:
This is an interesting case . . . Instead of disabling the rule here, I think a better solution is to change the code as follows:

```typescript
const file = await withTempFile((tmpFileName) => {
  expect(tmpFileName).toBeTruthy();
  return Promise.resolve(tmpFileName);
});
```

Our anonymous function here technically isn't `async` . . . but we were using that as a short-hand to say that the function returns a `Promise`. I think that returning an explicit `Promise` is probably a better approach.

Reply:
aww i see yeah withTempFile should return a promise. that make sense.